### PR TITLE
Fix repeated on_eos events from OpenAL players and cleanup the MediaThread

### PR DESCRIPTION
**Pull request by Square789**
See https://discord.com/channels/438622377094414346/1120364821288845332/1120364828645670973 for the original problem and why

### CODE REVIEW

File: pyglet/media/mediathreads.py [hunk 1]
```diff
@@ -10,143 +10,109 @@
...
+
+            assert _debug(f'PlayerWorkerThread: woke up @{time.time()}')
+            if self._stopped:
+                break
+
+            with self._operation_lock:
+                if self.players:
+                    sleep_time = self._nap_time
+                    for player in self.players:
+                        pass
```
[the remark this answers is not in the document]
Comment (Square789) [reply pyglet/media/mediathreads.py]: I appreciate the hint, steel automaton

**Review by benmoran56 — DISMISSED** [review]

**Comment by benmoran56** [comment]
Nice work on that cleanup. Let me know if you're ready to merge. 
I'll try this on my machine as well.

**Comment by Square789** [comment]
I just tried the directsound backend on a VM (should really have done that before) and it deadlocks, as those players may directly remove themselves from the thread. I will look into that some more but probably hold off on the merge for now

**Comment by Square789** [comment]
I probably should've hit that review rerequesting button this morning; oops
Directsound now has slightly different behavior on when it pauses its source, but that really just gets it into alignment with OpenAL. More importantly, it does not deadlock anymore and also does not produce the duplicate `on_eos` events anymore.

**Review by benmoran56 — APPROVED** [review]

**Comment by benmoran56** [comment]
This looks good! nice cleanup as well.

**Pull request merged**